config/v1/types_authentication.go

```diff
@@ -222,6 +222,10 @@ type OIDCProvider struct {
 	//
 	// +listType=atomic
 	ClaimValidationRules []TokenClaimValidationRule `json:"claimValidationRules,omitempty"`
+
+	// +optional
+	// userValidationRules provides the configuration for a single user validation rule.
```

Comment:
A couple things:
- Nit: Place the `// +optional` marker under the godoc for the field
- This GoDoc is pretty light on information. Take a look at https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#write-user-readable-documentation-in-godoc for some more information on how to write user readable godoc

Reply:
We also are going to want to include the appropriate server-side apply markers here